Sync server

rheinardkorf opened the pull request:
### Todo
- [x] Create a Sync Server
- [x] Verify parity
- [x] Setup local development
- [x] Test all three SQS queues
- [x] Update stdout to only include warnings/errors/failures to remove the noise when the sync is happening.
- [x] Finalize GCP cluster settings and look into running the Sync Server as a single container that has a persistent storage disk attached to store the DB.
- [x] Code review

closes #18, closes #20, closes #24

rheinardkorf committed:
Reworked sync-server

rheinardkorf committed:
Remove redundant interface.

rheinardkorf committed:
Merge branch 'develop' into sync-server

rheinardkorf committed:
Remove redundant interface.

rheinardkorf committed:
Add comments to wporg.go

rheinardkorf committed:
Fix odd dependencies introduced by IDE

rheinardkorf committed:
Comment code. Add forced sync.

rheinardkorf committed:
Move 'src' to 'wptide/pkg' and update refs.

rheinardkorf committed:
Update glide.lock

rheinardkorf committed:
Update glide.lock again

rheinardkorf committed:
Add error output for processes.

rheinardkorf committed:
Update glide.lock

rheinardkorf committed:
Update glide.lock

rheinardkorf commented:
@valendesigns I have opened the PR, but it will not be ready until we have added the containerisation.

valendesigns committed:
Setup Sync Server DevOps

valendesigns commented:
@rheinardkorf Tomorrow, could you troubleshoot why the sync server doesn't work when you bring the container up.

rheinardkorf committed:
Add CA Root Certs to sync-server image.

rheinardkorf commented:
@valendesigns it was the SSL root certs again.

I've added a command to the Makefile `sync.update.certs`.  It gets it from the `curl` mirror, but they request it not being done more than once a day, so I'm not including it in the build step.  Each time we're ready for deployment would be a good time to get new root certs, commit it to the repo, and rebuilt the image.

This gets the sync-server running for me now in the container.  Please review the changes.

valendesigns committed:
Rework ca-certificates & db storage

valendesigns committed:
Merge develop

valendesigns committed:
Change example db store path

valendesigns commented:
@rheinardkorf How do the latest changes look to you? I'm still testing a bit locally, but should be doing a more thorough review of the Go code tomorrow.

valendesigns committed:
Revert default value

rheinardkorf committed:
Remove "status" messages in favor of logging only errors.

Also fixed bug where an error should've been returned, but got a message
only.

Updated test.

valendesigns committed:
Ensure only one node & pod are always running

valendesigns committed:
Add Coveralls

valendesigns committed:
Change service name

valendesigns committed:
Add Coveralls badge

valendesigns commented on cmd/sync-server/main.go:
```diff
@@ -0,0 +1,261 @@
+			Active   bool
+			Accepts  string // "all" or "themes" or "plugins"
+		}{
+			"primary": {
```
I'm thinking that we should have two queues, a PHPCS queue and a Lighthouse queue. What purpose does this configuration serve? We don't have an immediate need for a plugin queue and being explicit about what the queue is makes more sense semantically to me than generic queues for themes and plugins.

valendesigns commented on cmd/sync-server/main.go:
```diff
@@ -0,0 +1,261 @@
+	for {
+		select {
+		case project := <-projects:
+			// If this project is not in sync (or forced), then sent it to the queue.
```
There is a typo in the comment `sent` should be `send`.

valendesigns replied:
fixed in [#610f777](https://github.com/xwp/go-tide/pull/23/commits/610f777988873bc6d2fb32688bd9c8dbfafdcc14)

valendesigns commented on cmd/sync-server/main.go:
```diff
@@ -0,0 +1,261 @@
+func defaultStandards() []string {
+	return []string{
+		"phpcs_wordpress",
+		"phpcs_phpcompatibility",
```
What about lighthouse?

rheinardkorf replied:
we can add that, but it should also be added below to `defaultAudits`...  once we sort out the audit server I'd like to review this.

valendesigns replied:
Does Lighthouse have any real options being passed to it?

valendesigns replied:
We'll double back on this once we've identified what the options would be.

valendesigns committed:
Fix spacing

valendesigns committed:
Fix typo

valendesigns committed:
Rework queues & variables

valendesigns committed:
Lower pool & raise delay
